### PR TITLE
Methylated GoldenGates

### PR DESCRIPTION
**Pull request by Koeng101**
## Changes in this PR
I add the ability to use methylated sequences in GoldenGates

### Why are you making these changes?
I need a way to simulate methylated GoldenGates

### Are any changes breaking? (IMPORTANT)
Kinda? But IMO worth and not sure anyone else is using.

## Pre-merge checklist

* [x] New packages/exported functions have docstrings.
* [x] New/changed functionality is thoroughly tested.
* [x] New/changed functionality has a function giving an example of its usage in the associated test file. See `primers/primers_test.go` for what this might look like.
* [x] Changes are documented in `CHANGELOG.md` in the `[Unreleased]` section.
* [x] All code is properly formatted and linted.
* [x] The PR template is filled out.

### CODE REVIEW

**Comment by Koeng101** [comment]
Fixes #390

**Comment by TimothyStiles** [comment]
@Koeng101 is this ready to merge after conflicts are resolved?

**Comment by Koeng101** [comment]
Good lord that last PR had a LOT of conflicts. Working on it, but basically yea its ready after those are ironed out

**Comment by Koeng101** [comment]
Ready!

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,9 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added a parser and writer for the `pileup` sequence alignment format (#329)
 - Added statistics to the `synthesis/codon` package (keeping track of the observed start codon occurrences in a translation table) (#350)
 - Added option to fragmenter to fragment with only certain overhangs (#387)
-
-
-
+- GoldenGates with methylated sequences using lowercase letters (#391)
```
Comment (TimothyStiles) [inline CHANGELOG.md:15]: include note on function signature including what I think is a methylation flag in the clone function @Koeng101?

Comment (Koeng101) [reply CHANGELOG.md:15]: what should I add to the following documentation?

https://github.com/TimothyStiles/poly/blob/debc0604d0696987c634c3cceff9876d581b4cdc/clone/clone.go#L345-L347

**Pull request closed without merging**